impedance/models/circuits/elements.py

```diff
@@ -1,27 +1,27 @@
 import numpy as np
 
-initial_state = globals().copy()
-non_element_functions = ['element_metadata',
-                         'initial_state',
-                         'non_element_functions',
-                         'typeChecker',
-                         'circuit_elements']
-# populated at the end of the file -
-# this maps ex. 'R' to the function R to always give us a list of
-# active elements in any context
-circuit_elements = {}
+
+class ElementException(Exception):
+    ...
+
+
+class OverWriteElementException(ElementException):
+    ...
 
 
-def element_metadata(num_params, units):
-    """ decorator to store metadata for a circuit element
+def element(num_params, units, overwrite=False):
+    """decorator to store metadata for a circuit element
 
-        Parameters
-        ----------
-        num_params : int
-            number of parameters for an element
-        units : list of str
-            list of units for the element parameters
+    Parameters
+    ----------
+    num_params : int
+        number of parameters for an element
+    units : list of str
+        list of units for the element parameters
+    overwrite : bool (default False)
+        whether or not you're overwriting an already existing element
```

Comment:
Tiny suggestion, but the `whether or not` was a little confusing to me. Could we make it clearer what overwrite=True does with something like `if true, overwrites any existing element...`

```suggestion
        if true, overwrites any existing element; if false, raises OverwriteError if element name already exists.
```